tests/unit/test_charm.py

```diff
@@ -338,3 +355,185 @@ def test_given_cannot_connect_to_container_when_nrf_available_then_status_is_wai
             state_out.deferred[0].name,
             "nrf_available",
         )
+
+    @patch("charm.generate_private_key")
+    @patch("ops.model.Container.push")
+    def test_given_can_connect_when_on_certificates_relation_created_then_private_key_is_generated(
+        self, patch_push, patch_generate_private_key
+    ):
+        private_key = b"private key content"
+        patch_generate_private_key.return_value = private_key
+        state_in = State(
+            leader=True,
+            containers=[self.container],
+            relations=[self.nrf_relation, self.tls_relation],
+        )
+
+        self.ctx.run(self.tls_relation.created_event, state_in)
+
+        patch_push.assert_called_with(
+            path="/free5gc/support/TLS/ausf.key", source=private_key.decode()
+        )
```

Comment:
With scenario, it is not necessary to patch `Container.push`. The test `test_given_relations_created_and_nrf_data_available_when_pebble_ready_then_config_file_rendered_and_pushed` contains an example on how to do this assertion through a `mount`.

Reply:
This goes for all the filesystem operations that are patched in this file.

Reply:
Done

Reply:
I am not sure if the changes were overwritten, I still see the old code with `patch_push`.

Reply:
Good catch, not sure how they left but I re-added the changes.